### PR TITLE
feat: add `Partly Paid` status in Invoices

### LINKED ISSUE #26969: Mark Sales Invoice as Overdue when first payment becomes overdue
**Is your feature request related to a problem? Please describe.**

In **Sales Invoice** we can specify a payment schedule. For example, 50 % after 14 days, 50 % after 30 days. Currently the latest date in the payment schedule is used as the invoice's due date. Only when the last payment becomes overdue, the invoice status changes to "Overdue".

However, we regard a **Sales Invoice** as overdue based on the scheduled payments. Once the first payment is overdue, we want to create a **Dunning**, reminding the customer that he is late on this payments.

**Describe the solution you'd like**

- The status of a **Sales Invoice** depends on the payment schedule instead of due date
- Set status to "Overdue" once a scheduled payment is overdue
- Introduce a new status "Partially Paid" that is displayed when payments have been received according to the payment schedule.
- ~Extend **Dunning** to consider only overdue payments and calculate overdue days correctly. Probably it would make sense to copy the **Sales Invoice**'s payment schedule to the **Dunning**.~ -> Create a new Issue for this

**Describe alternatives you've considered**
> A clear and concise description of any alternative solutions or features you've considered.

**Additional context**

The option "Allocate Payment Based On Payment Terms" is set in **Payment Terms Template**.

### PR DESCRIPTION
**Pull request by pruthvi145**
- Resolves https://github.com/frappe/erpnext/issues/26969
- Also uses `getdate()` instead of `CURDATE()` to get accurate dates based on timezones set in System Settings.

## Behaviour
If a part of the invoice amount has been received, `Partly Paid` state will be shown (instead of `Unpaid`). If any scheduled payment (as specified in the Payment Schedule table) is pending beyond it's due date, the `Overdue` state will be shown.

## Documentation
https://docs.erpnext.com/docs/v13/user/manual/en/accounts/purchase-invoice/edit?wiki_page_patch=a8b85aba51
https://docs.erpnext.com/docs/v13/user/manual/en/accounts/sales-invoice/edit?wiki_page_patch=709cb5df15

### CODE REVIEW

**Comment by olamide2** [comment]
cool

**Comment by ankush** [comment]
@pruthvi145 please sort the imports with `isort` or setup pre-commit: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#linting 

You can ignore errors that are not from your change... we will take care of them 😅

**Comment by pruthvi145** [comment]
> @pruthvi145 please sort the imports with `isort` or setup pre-commit: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist#linting
> 
> You can ignore errors that are not from your change... we will take care of them 😅

Done! I have sorted for all files(2).

**Review by barredterra — COMMENTED** [review]
Tested, looks good to me. Thanks for your work, @pruthvi145!

I think we should add a unit test for Partly Paid status to make this PR complete.

**Review by barredterra — DISMISSED** [review]

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.json [hunk 1]
```diff
@@ -726,7 +725,6 @@
    "read_only": 1
   },
   {
-   "depends_on": "packed_items",
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice.json]: These seem to be unintentional changes? Pull `develop` branch changes then migrate & add the new status.

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.py [hunk 2]
```diff
@@ -1507,12 +1498,42 @@ def set_status(self, update=False, status=None, update_modified=True):
 					self.status = "Paid"
 				else:
 					self.status = "Submitted"
+
+				if (
+					self.status in ("Unpaid", "Partly Paid", "Overdue")
+					and self.is_discounted
+					and get_discounting_status(self.name) == "Disbursed"
+				):
+					self.status += " and Discounted"
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice.py:1507]: There's no "Partly Paid and Discounted" option added in the sales invoice status.

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.json (hunk 1 above)
Comment (sagarvora) [reply erpnext/accounts/doctype/sales_invoice/sales_invoice.json]: Yes. Thanks!

File: erpnext/accounts/doctype/purchase_invoice/purchase_invoice.json [hunk 3]
```diff
@@ -627,48 +533,37 @@
...
-   "show_days": 1,
-   "show_seconds": 1
+   "print_hide": 1
   },
   {
    "fieldname": "scan_barcode",
    "fieldtype": "Data",
-   "label": "Scan Barcode",
-   "options": "Barcode"
+   "label": "Scan Barcode"
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/purchase_invoice/purchase_invoice.json:566]:
```suggestion
   "label": "Scan Barcode",
   "options": "Barcode"
```

File: erpnext/accounts/doctype/sales_invoice/sales_invoice_list.js [hunk 4]
```diff
@@ -15,6 +15,7 @@ frappe.listview_settings['Sales Invoice'] = {
 			"Unpaid and Discounted": "orange",
 			"Overdue and Discounted": "red",
 			"Overdue": "red",
+			"Partly Paid": "yellow",
```
Comment (nextchamp-saqib) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice_list.js:18]:
```suggestion
			"Partly Paid": "yellow",
			"Partly Paid and Discounted": "yellow",
```

**Review by nextchamp-saqib — APPROVED** [review]

**Pull request closed without merging**